.github/workflows/linter.yml

```diff
@@ -0,0 +1,51 @@
+---
+###########################
+###########################
+## Linter GitHub Actions ##
+###########################
+###########################
+name: Lint Code Base
+
+#
+# Documentation:
+# https://help.github.com/en/articles/workflow-syntax-for-github-actions
+#
+
+#############################
+# Start the job on all push #
+#############################
+on:
+  push:
+    branches-ignore:
+      - main
```

Comment:
The comment does not explain why `main` is excluded, it's not _all push_ events. Also based on the commit history, it does not seem to me that pull requests are enforced in this repository, so I would not exclude the default branch from linting.

Reply:
The `main` is excluded, because whole linting/testing process should be done only when working with PRs (non "main" branches).

I doesn't make sense to lint/test anything in main, because in such case the change is already there and can not be easily reverted/change comparing to PRs... 

Example: https://github.com/github/super-linter#example-connecting-github-action-workflow

Reply:
I see, but it does not seem to me that PRs are mandated (@ondj ?). If they are not, the default branch should be linted as well. Otherwise if any commit breaks the linting in the default branch, you won't know until someone creates a PR, where the linter will fail, which creates bad experience for contributors. If direct push to the default branch is allowed, you show know about failing CI as soon as possible, so you can fix it.

Reply:
Ok... Changed...